### PR TITLE
move copy/pasted limit()/offset() code to traits

### PR DESCRIPTION
**Pull request by pavarnos**
similar to #116: the code for limit() and offset() was duplicated in several classes. Extracted the common code to traits, added getLimit() and getOffset() to the relevant interfaces.

### CODE REVIEW

**Comment by pavarnos** [comment]
This PR is not as smooth and simple as #116 because Common\Select::page/paging has complex interactions with limit/offset. 
But: note the improvement in code quality: Scrutinizer reports 9.34 (up from 8.8?), mostly influenced by Common\Select

File: src/AbstractQuery.php [hunk 1]
```diff
@@ -491,22 +473,6 @@ protected function buildOrderBy()
      */
     protected function buildLimit()
     {
-        $clause = '';
```
Comment (harikt) [inline src/AbstractQuery.php:494]: if this method just returns an empty string what is the value for it ?

Comment (pavarnos) [reply src/AbstractQuery.php:494]: as it says in the comment: it is a template method https://en.wikipedia.org/wiki/Template_method_pattern "one or more algorithm steps can be overridden by subclasses to allow differing behaviors while ensuring that the overarching algorithm is still followed"
The buildLimit() is here to hold its place in the build() algorithm for a derived method to do something useful with.

Comment (pavarnos) [reply src/AbstractQuery.php:494]: It is needed by Delete and Update

**Pull request merged**